proposals/0002-Extracted-Command-Line-Tools.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Extract CLI from Core
+author:
+- Mike Grabowski
+date: 2018-08-10
+---
+
+# RFC0000: Extract CLI from Core
+
+## Summary
+
+Extract the current CLI implementation from React Native Core and move it into a third party package.
+
+## Basic example
+
+Command line tools in React Native are split into two, separate pieces. 
+
+First one is the `react-native-cli` package that we prompt users to install globally, for the first time they actually interact with the framework. This package, by the design, is not meant to be frequently updated, which eliminates many annoying issues that global packages are infamous for.
+
+The second one is `local-cli` package which is the actual source code of the CLI. It is a set of multiple commands, each of them grouped into a standalone module, that are loaded altogether when the CLI is spawned. 
+
+The biggest part of the `local-cli` is its `core`, which is capable of analyzing your source code. It provides a context object for all the commands, that contains necessary information about the location of your source files, available platforms and more. Commands such as `link`, `unlink` or `run-ios` use that information to perform their work.
+
+This proposal suggests no changes to the end user in its first iteration. We would simply couple new package with React Native version, just like we do with Metro. That makes it possible to still make React Native version specific changes to the CLI itself. 
+
+## Motivation
+
+Facebook does not use CLI provided with React Native, which makes it hard to maintain the project. There is no clear owner and issues / pull requests get stack in the sea of other, non-CLI related reports.
+
+It is also hard to keep moving on the CLI as every additional dependency or code change is subject to the heavy review process and sync.
+
+Having a separate package with me (and Callstack) being strategic maintainers would allow us to iterate on the design of CLI even more and make sure we have the best tooling out there.
+
+Now, if you have been following React Native development for a few years, you know I have been always super excited about CLI in general. I did many talks in that field and lead efforts of creating RNPM and then, merging it to the core as `react-native link` together with Alexey Kureev. Link, and RNPM, in general, was just the first step on our way to meet that high expectation of high-quality tools. Unfortunately, due to the aforementioned constraints and lack of time we were never able to complete our long-term strategy.
```

Comment:
> Now, if you have been following React Native development for a few years, you know I have been always super excited about CLI in general

I think it might be best to keep anecdotal things out of the context of these proposals, so that folks don't need a history lesson or feel they need to know someone in the industry to participate.

While I appreciate you're excited by this (and I know you'll be a great maintainer!), I'm not sure your personality contributes to the technical impact that this will have. 😄

Reply:
Sure, thanks for leaving this feedback. I guess it's due to my understanding of "motivation" that I took a bit personal as well, rather than focusing on "technical motivation" in the first place.